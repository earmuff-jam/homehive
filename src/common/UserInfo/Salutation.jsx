import { Stack, Typography } from "@mui/material";

export default function Salutation({ userInfo, isEnd = false }) {
  return (
    <Stack sx={{ my: 4 }}>
      <Typography variant="subtitle2" sx={{ color: "#666" }}>
        {isEnd ? "Thank you," : "To,"}
      </Typography>
      <Stack direction="row" spacing={0.5}>
        <Typography
          variant="subtitle2"
          sx={{ color: "#666", textTransform: "capitalize" }}
        >
          {userInfo.first_name}
        </Typography>
        <Typography
          variant="subtitle2"
          sx={{ color: "#666", textTransform: "capitalize" }}
        >
          {userInfo.last_name}
        </Typography>
      </Stack>
      <Typography
        variant="subtitle2"
        sx={{ color: "#666", textTransform: "capitalize" }}
      >
        {userInfo.street_address}
      </Typography>
      <Typography
        variant="subtitle2"
        sx={{ color: "#666", textTransform: "capitalize" }}
      >
        {userInfo.city} {userInfo.state}, {userInfo.zipcode}
      </Typography>
    </Stack>
  );
}
