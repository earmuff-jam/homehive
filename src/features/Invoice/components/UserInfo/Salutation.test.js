import React from "react";
import { render, screen } from "@testing-library/react";
import Salutation from "./Salutation";

describe("Salutation component", () => {
  const mockUser = {
    first_name: "John",
    last_name: "Doe",
    street_address: "123 Main St",
    city: "Austin",
    state: "TX",
    zipcode: "78701",
  };

  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(<Salutation userInfo={mockUser} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders 'To,' by default", () => {
    render(<Salutation userInfo={mockUser} />);
    expect(screen.getByText("To,")).toBeInTheDocument();
  });

  it("renders 'Thank you,' when isEnd=true", () => {
    render(<Salutation userInfo={mockUser} isEnd />);
    expect(screen.getByText("Thank you,")).toBeInTheDocument();
  });

  it("renders all user info fields correctly", () => {
    render(<Salutation userInfo={mockUser} />);
    expect(screen.getByText("John")).toBeInTheDocument();
    expect(screen.getByText("Doe")).toBeInTheDocument();
    expect(screen.getByText("123 Main St")).toBeInTheDocument();
    expect(screen.getByText("Austin TX, 78701")).toBeInTheDocument();
  });
});
