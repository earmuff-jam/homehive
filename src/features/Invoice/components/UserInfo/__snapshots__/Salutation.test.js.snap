// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Salutation component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-u6jtrn-MuiStack-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-1x0mq5c-MuiTypography-root"
    >
      To,
    </h6>
    <div
      class="MuiStack-root css-69ywkl-MuiStack-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-1x0mq5c-MuiTypography-root"
      >
        John
      </h6>
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-1x0mq5c-MuiTypography-root"
      >
        Doe
      </h6>
    </div>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-1x0mq5c-MuiTypography-root"
    >
      123 Main St
    </h6>
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 css-1x0mq5c-MuiTypography-root"
    >
      Austin TX, 78701
    </h6>
  </div>
</DocumentFragment>
`;
